### ISSUE TITLE
Switching Time Periods Shows Error

### ISSUE THREAD

**Issue by Final-Hawk** [jokob-sk/Pi.Alert#42]
If you try to switch the time period using the drop down on the device details page, an error occurs. The setting applies and works after a refresh of the page, but produces an error and does not update the page live. See attached screenshot.

Steps to reproduce:
1. Go to a devices details page
2. Use the drop down in the top right to change the time period. For example, from today to last month.
3. Browser gives error and does not update the page information accordingly.
4. Refresh page, and the selected period setting shows with the correct information.
Seems to just be an error with updating the page live, not an issue updating the period on the back-end.  Hope this makes sense and is reproducible. 
This is not 100% reproducible on the first try, but it is if you continue to select different time periods. Seems to be random. 
![image](https://user-images.githubusercontent.com/63589370/180637756-946fcb8d-57da-40fa-aefa-630af8aea17c.png)
Somewhat related in the screenshot, presence time is 0 hours because of #30

**Comment by jokob-sk** [comment]
Probably related:

![image](https://i.imgur.com/DDNIGkP.png)

**Comment by Final-Hawk** [comment]
Swapping that line out with
```
  if ($row[0] == null) {
    $deviceData['dev_Sessions'] = 1;
  } else {
    $deviceData['dev_Sessions'] = $row[0];
  }
```
Made the session count 1 if it was 0 before, but the presence data and graph does not change and i still have the issue when switching between times. I get different errors in the console though, just 500 errors, i have attached a screenshot. 

![image](https://user-images.githubusercontent.com/63589370/180641587-4fcc87d5-d387-4485-ba20-23a0d45ba762.png)
After clicking OK on that error, this error message follows it
![image](https://user-images.githubusercontent.com/63589370/180641627-6b0024ab-9482-47a8-ace7-e63bfed34280.png)

**Comment by jokob-sk** [comment]
I was doing some debugging on my end and I still think the below is necessary:

```php
  if ($row[0] == null) {
    $deviceData['dev_Sessions'] = 1;
  } else {
    $deviceData['dev_Sessions'] = $row[0];
  }
```

Also, I think the 500 errors are only showing up if you switch too fast and the previous request wasn't finished yet. If you wait a few seconds between switching the time ranges (2-3s) the switching seems to work fine. 
 
If that's the case we need to implement a spinning wheel / loading screen or something when switching the ranges until the view if refreshed.

I have another theory - it might be also caused by too frequent UI refreshes (the server timestamp with the continuously updating seconds - but I haven't tried removing it - have to wrap up for now).

**Comment by jokob-sk** [comment]
I have also a new theory - I think the dropdown (1) is only supposed to work for the Sessions tab. The Presence tab has its' own dedicated year | month | week (2) selector:

![img](https://i.imgur.com/KdVzism.png)

The dropdown works in the Sessions tab at least (with the occasional error being thrown). 

What are your thoughts?

**Comment by Final-Hawk** [comment]
I don't think that's the issue, for what i can see the presence tab does not read that period data. See /front/php/server/events.php line 259. It reads the variable but does not use it.
I think it could be because the database seems to get locked randomly, see the lighttpd logs in the screenshot below. Arpscan was off, so it wasnt adding new devices. 
![image](https://user-images.githubusercontent.com/63589370/180737739-5bcff676-598f-4e5a-bc1a-c8b80e895e35.png)


![image](https://user-images.githubusercontent.com/63589370/180737031-571d407c-db16-4014-8beb-7b2b08372138.png)

**Comment by jokob-sk** [comment]
I'm not a PHP/DB expert, but I assume that HTTP 500 errors are thrown because the previous request didn't finish and the previous request accesses the DB, which is locked for the request. So tracking back, the issue is still that the previous request didn't finish. 

Mitigation would be to lock the front end until previous request finish (an issue is also the CRON process I assume - maybe having it executed every 5 min (compared to the previous 15 min) aggravated the issue. 

Not sure if SQLite has the ability to execute queries with a 'NO LOCK' directive (at least the read queries).

I'm also looking at this thread: https://stackoverflow.com/questions/6101480/sqlite-3-locking-in-php

**Comment by Final-Hawk** [comment]
Closed with #47
